dds/DCPS/RTPS/Spdp.h

```diff
@@ -339,7 +339,15 @@ class OpenDDS_Rtps_Export Spdp : public DCPS::LocalParticipant<Sedp>
     void execute()
     {
       if (action_ == CMG_JOIN) {
-        tport_->join_multicast_group(nic_, all_interfaces_);
+        bool success = false;
+        for (int count = 0; count <= 1 && !success; ++count) {
+          success = tport_->join_multicast_group(nic_, all_interfaces_);
```

Comment:
This just calls the join again? Does this actually fix the broken behavior, or are we crossing our fingers here and hoping the second time behaves better than the first? At the best, this seems it would reduce a N% failure rate to a SQRT(N)% failure rate.

It would be good to understand why the failure happened in these cases so we can actually prevent it if possible. Have you been able to catch the error in a debugger? Was the interface name wrong? Does this version of Windows / Linux need a special flag or system call to prevent interruption? Or perhaps we need to check for a special return status in order to know when to try again?

Reply:
The problem is that you have multiple transports on different threads hitting the join at the same time.  As far as I can tell this happens only in windows 7, but it might be related to the failed to join multicast group issue also appearing in fedora.  

Barring a lock which would only be useful for windows 7, while causing a slowdown for other OS, Adam and I agreed this was probably the least impactful solution.